front/src/modules/trainschedule/components/DriverTrainScheduleV2/__tests__/utils.spec.ts

```diff
@@ -0,0 +1,30 @@
+import { findActualVmax } from '../utils';
+
+describe('findActualVmax', () => {
+  const vMax = { internalBoundaries: [2000, 3400, 5300, 6000], speeds: [10, 100, 200, 100, 150] };
+
+  it('should return the correct Vmax', () => {
+    const result = findActualVmax(2500, vMax);
+    expect(result).toEqual(100);
+  });
+
+  it('should return the correct Vmax when the givenPosition is in the last interval', () => {
+    const result = findActualVmax(7000, vMax);
+    expect(result).toEqual(150);
+  });
+
+  it('should return the correct Vmax when the givenPosition is in thefirst interval', () => {
+    const result = findActualVmax(1000, vMax);
+    expect(result).toEqual(10);
+  });
+
+  it('should return the min Vmax when givenPosition is equal to a boundary (min before)', () => {
+    const result = findActualVmax(3400, vMax);
+    expect(result).toEqual(100);
+  });
+
+  it('should return the min Vmax when givenPosition is equal to a boundary (min after)', () => {
+    const result = findActualVmax(5300, vMax);
+    expect(result).toEqual(100);
```

Comment:
Thanks for those tests!
Added a test on both types of boudaries (min before or after), and also put first speed at `10` to avoid cases where `0` value would be a hidden bug (as it may be the default value).
Done in https://github.com/OpenRailAssociation/osrd/pull/8351/commits/228d679c54799f5835bfe6d8f0a6e090c1c9b2c4